src/components/ReportActionItem/MoneyRequestView.js

```diff
@@ -190,12 +200,13 @@ function MoneyRequestView({report, parentReport, parentReportActions, policyCate
                         </View>
                     </OfflineWithFeedback>
                 )}
-                {!hasReceipt && canEdit && !isSettled && canUseViolations && (
+                {!hasReceipt && canEdit && !isSettled && (
                     <ReceiptEmptyState
                         hasError={hasErrors}
                         onPress={() => Navigation.navigate(ROUTES.EDIT_REQUEST.getRoute(report.reportID, CONST.EDIT_REQUEST_FIELD.RECEIPT))}
                     />
                 )}
+                {canUseViolations && <FieldViolationMessages violations={getViolationsForField('receipt')} />}
```

Comment:
All the occurrences of `canUseViolations` should be function calls `canUseViolations()` right?
```suggestion
                {canUseViolations() && <FieldViolationMessages violations={getViolationsForField('receipt')} />}
```

Reply:
I took a look and apparently not. Use permissions returns a map from keys to boolean values:

```
export default function usePermissions(): UsePermissions {
    const betas = useContext(BetasContext);
    return useMemo(() => {
        const permissions: UsePermissions = {};

        for (permissionKey in Permissions) {
            if (betas) {
                const checkerFunction = Permissions[permissionKey];

                permissions[permissionKey] = checkerFunction(betas);
            }
        }

        return permissions;
    }, [betas]);
}
```

I think `canUseViolations()` is one of the checker functions, so the `usePermissions` hook calls it and stores the result in that map with a key that has the same name as the function.

If you highlight `canUsePermissions` and choose 'go to definition' it takes you to the function, but that's not the actual value being returned there.

Reply:
Nice, thanks for the clarification